modules/apps/commerce/commerce-shipping-engine/src/main/java/com/liferay/commerce/shipping/engine/internal/taglib/FunctionCommerceShippingMethodConfigurationScreenNavigationEntry.java

```diff
@@ -0,0 +1,173 @@
+ */
+@Component(
+	property = "screen.navigation.entry.order:Integer=10",
+	service = ScreenNavigationEntry.class
+)
+public class FunctionCommerceShippingMethodConfigurationScreenNavigationEntry
+	implements ScreenNavigationEntry<CommerceShippingMethod> {
+
+	@Override
+	public String getCategoryKey() {
+		return FunctionCommerceShippingScreenNavigationConstants.
+			CATEGORY_KEY_FUNCTION_COMMERCE_SHIPPING_METHOD_CONFIGURATION;
+	}
+
+	@Override
+	public String getEntryKey() {
+		return getCategoryKey();
+	}
+
+	@Override
+	public String getLabel(Locale locale) {
+		return language.get(locale, "configuration");
+	}
+
+	@Override
+	public String getScreenNavigationKey() {
+		return FunctionCommerceShippingScreenNavigationConstants.
+			SCREEN_NAVIGATION_KEY_COMMERCE_SHIPPING_METHOD;
+	}
+
+	@Override
+	public boolean isVisible(
+		User user, CommerceShippingMethod commerceShippingMethod) {
+
+		if (commerceShippingMethod == null) {
+			return false;
+		}
+
+		CommerceShippingEngine commerceShippingEngine =
+			_commerceShippingEngineRegistry.getCommerceShippingEngine(
+				commerceShippingMethod.getEngineKey());
+
+		return commerceShippingEngine instanceof FunctionCommerceShippingEngine;
+	}
+
+	@Override
+	public void render(
+			HttpServletRequest httpServletRequest,
+			HttpServletResponse httpServletResponse)
+		throws IOException {
+
+		try {
+			long commerceChannelId = ParamUtil.getLong(
+				httpServletRequest, "commerceChannelId");
+
+			httpServletRequest.setAttribute(
+				"commerceChannelId", commerceChannelId);
+
+			String commerceShippingMethodEngineKey = ParamUtil.getString(
+				httpServletRequest, "commerceShippingMethodEngineKey");
+
+			httpServletRequest.setAttribute(
+				"commerceShippingMethodEngineKey",
+				commerceShippingMethodEngineKey);
+
+			long commerceShippingMethodId = ParamUtil.getLong(
+				httpServletRequest, "commerceShippingMethodId");
+
+			httpServletRequest.setAttribute(
+				"commerceShippingMethodId", commerceChannelId);
+
+			CommerceShippingMethod commerceShippingMethod =
+				_commerceShippingMethodLocalService.getCommerceShippingMethod(
+					commerceShippingMethodId);
+
+			UnicodeProperties typeSettingsUnicodeProperties =
+				commerceShippingMethod.getTypeSettingsUnicodeProperties();
+
+			if (typeSettingsUnicodeProperties.isEmpty()) {
+				FunctionCommerceShippingEngine functionCommerceShippingEngine =
+					(FunctionCommerceShippingEngine)
+						_commerceShippingEngineRegistry.
+							getCommerceShippingEngine(
+								commerceShippingMethodEngineKey);
+
+				httpServletRequest.setAttribute("isDefaultValue", Boolean.TRUE);
+				httpServletRequest.setAttribute(
+					"shippingMethodTypeSettings",
+					functionCommerceShippingEngine.
+						getTypeSettingsUnicodeProperties());
+			}
+			else {
+				httpServletRequest.setAttribute(
+					"isDefaultValue", Boolean.FALSE);
+				httpServletRequest.setAttribute(
+					"shippingMethodTypeSettings",
+					typeSettingsUnicodeProperties);
```

Comment:
@luca-pellizzon request attributes need to be `*WebKeys*`, https://github.com/brianchandotcom/liferay-portal/pull/144387#issuecomment-1827651028